conu/apidefs/container.py

```diff
@@ -0,0 +1,234 @@
+        :return: instance of Container
+        """
+        raise NotImplementedError("run_via_binary method is not implemented")
+
+    @classmethod
+    def run_via_api(cls, image, container_params):
+        """
+        create container using provided image and run it in the background
+
+        :param image: instance of Image
+        :param container_params: instance of ContainerParameters
+        :return: instance of Container
+        """
+        raise NotImplementedError("run_via_api method is not implemented")
+
+    @classmethod
+    def create(cls, image, container_params):
+        """
+        create container using provided image
+
+        :param image: instance of Image
+        :param container_params: instance of ContainerParameters
+        :return: instance of Container
+        """
+        raise NotImplementedError("create method is not implemented")
+
+    def start(self):
+        """
+        start current container
+        :return: None
+        """
+        raise NotImplementedError("start method is not implemented")
+
+    def exec(self, command, **kwargs):
+        """
+        TODO: what about parameters?
+
+        :param command: command to execute in the container
+        :param kwargs:
+        :return: ? we need to provide output, exit code and there needs to be a possiblity for
+                  this thing to be async and blocking
+        """
+        raise NotImplementedError("exec method is not implemented")
+
+    def logs(self, follow=False):
+        """
+        get logs from this container
+
+        :param follow: bool, provide iterator if True
+        :return: str or iterator
+        """
+        raise NotImplementedError("logs method is not implemented")
+
+    def stop(self):
+        """
+        stop this container
+
+        :return: None
+        """
+        raise NotImplementedError("stop method is not implemented")
+
+    def rm(self, force=False):
+        """
+        remove this container
+
+        :param force: bool, if container engine supports this, force the functionality
+        :return: None
+        """
+        raise NotImplementedError("rm method is not implemented")
+
+    def copy_to(self, src, dest):
+        """
+        copy a file or a directory from host system to a container
+
+        :param src: str, path to a file or a directory on host system
+        :param dest: str, path to a file or a directory within container
+        :return: None
+        """
+        raise NotImplementedError("copy_to method is not implemented")
+
+    def copy_from(self, src, dest):
+        """
+        copy a file or a directory from container to host system
+
+        :param src: str, path to a file or a directory within container
+        :param dest: str, path to a file or a directory on host system
+        :return: None
+        """
+        raise NotImplementedError("copy_from method is not implemented")
+
+    def read_file(self, file_path):
```

Comment:
Would it be better to have a `File` object (that we'd also implement in conu) that could:

* be used as a context manager (e.g. `with file.read():`)
* be read in text or binary mode
* would contain all other info, like creation timestamps, permissions, etc?

Maybe I'm overengineering this, but I think it'd be nice :)

Reply:
so this be `def get_file(self, file_path):`? I like it.

Reply:
Yes, exactly. And then users could do e.g.
```
with container.get_file('/usr/bin/foo').open('rb') as f:
    do_something()
```

Reply:
Love it!